agent/consul/rate/handler.go

```diff
@@ -0,0 +1,103 @@
+// package rate implements server-side RPC rate limiting.
+package rate
+
+import (
+	"context"
+	"net"
+	"sync/atomic"
+
+	"github.com/hashicorp/consul/agent/consul/rate/multilimiter"
+)
+
+// Mode determines the action that will be taken when a rate limit has been
+// exhausted (e.g. log and allow, or reject).
+type Mode int
+
+const (
+	// ModePermissive causes the handler to log the rate-limited operation but
+	// still allow it to proceed.
+	ModePermissive Mode = iota
+
+	// ModeEnforce causes the handler to reject the rate-limted operation.
+	ModeEnforce
```

Comment:
This comment is not particular to this PR but is a small nit that spans the RFCs that is hopefully a quick discussion and does not devolve into "Naming Things Is Hard!".

TLDR: I'd like to change "enforce" to "enforced" so both modes are adjectives.

LONGER CONTEXT:
The meat RFC refers to Mode and Action but eventually settles on Mode. To cut to the chase, Mode should have adjectives - permissive, strict, enforced, etc. If we named this action, the values would be verbs- enforce, permit, allow, etc.

I think the simple enforce -> enforced is easy and does not open the door too much.

Reply:
The lexicologie in the RFC was based on the selinux terms. I double checked what selinux use for the config and found out it actually use:
- disabled
- permissive
- enforcing

I'm not sure what is the reasoning behind that, If any, but I thought I would mention it. 

That said I'm good with naming this enforced too.

Reply:
Nice! Have changed to `enforcing`.